services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/QinQFlowSpec.groovy

```diff
@@ -0,0 +1,365 @@
+        given: "A switch with enabled multiTable mode"
+        def sw = topology.activeSwitches.find { northbound.getSwitchProperties(it.dpId).multiTable } ?:
+                assumeTrue("No suiting switches found", false)
+
+        when: "Create a single switch QinQ flow"
+        def flow = flowHelperV2.singleSwitchFlow(sw)
+        flow.source.vlanId = srcVlanId
+        flow.source.innerVlanId = srcInnerVlanId
+        flow.destination.vlanId = dstVlanId
+        flow.destination.innerVlanId = dstInnerVlanId
+        def response = flowHelperV2.addFlow(flow)
+
+        then: "Response contains correct info about innerVlanIds"
+        with(response) {
+            it.source.vlanId == (srcVlanId ? srcVlanId : srcInnerVlanId)
+            it.source.innerVlanId == (srcVlanId ? srcInnerVlanId : 0)
+            it.destination.vlanId == (dstVlanId ? dstVlanId : dstInnerVlanId)
+            it.destination.innerVlanId == (dstVlanId ? dstInnerVlanId : 0)
+        }
+
+        and: "Flow is really created with requested innerVlanIds"
+        with(northbound.getFlow(flow.flowId)) {
+            it.source.vlanId == (srcVlanId ? srcVlanId : srcInnerVlanId)
+            it.source.innerVlanId == (srcVlanId ? srcInnerVlanId : 0)
+            it.destination.vlanId == (dstVlanId ? dstVlanId : dstInnerVlanId)
+            it.destination.innerVlanId == (dstVlanId ? dstInnerVlanId : 0)
+        }
+
+        //is not implemented yet
+//        and: "Flow is valid and pingable"
+//        northbound.validateFlow(flow.flowId).each { assert it.asExpected }
+//        verifyAll(northbound.pingFlow(flow.flowId, new PingInput())) {
+//            it.forward.pingSuccess
+//            it.reverse.pingSuccess
+//        }
+
+        //is not implemented yet
+//        and: "Involved switches pass switch validation"
+//        with(pathHelper.getInvolvedSwitches(pathHelper.convert(northbound.getFlowPath(flow.flowId)))) {
+//            def validationInfo = northbound.validateSwitch(it.dpId)
+//            validationInfo.verifyRuleSectionsAreEmpty(["missing", "excess", "misconfigured"])
+//            validationInfo.verifyMeterSectionsAreEmpty(["missing", "excess", "misconfigured"])
+//        }
+
+        when: "Delete the flow"
+        flowHelperV2.deleteFlow(flow.flowId)
+
+        then: "Flow rules are deleted"
+        //is not implemented yet
+//        Wrappers.wait(RULES_INSTALLATION_TIME, 1) {
+//            assert northbound.getSwitchRules(sw.dpId).flowEntries*.cookie.sort() == sw.defaultCookies.sort()
+//        }
+
+        where:
+        srcVlanId | srcInnerVlanId | dstVlanId | dstInnerVlanId
+        10        | 20             | 30        | 40
+        10        | 20             | 30        | 0
+        10        | 20             | 0         | 0
+        10        | 0              | 0         | 40
+        0         | 20             | 30        | 0
+    }
```

Comment:
please check single switch single port flow too, it is a separate class of flows

Reply:
I will do it

Reply:
added